### PR TITLE
Headslug for everyone + Headslug once per absorb + Cling abilities offline after reviving

### PR DESCRIPTION
**Pull request by HMBGERDO**
## What Does This PR Do

Makes headslug default changeling ability meaning you dont need to purchase it.
Makes headslug available only once per absorption. You need to absorb target with mind to use it again.
Changeling abilities cannot be used 10 seconds after reviving with stasis.

## Why It's Good For The Game

Headslug in current form is OP. Its nearly impossible to kill changeling that uses headslug all the time. You need to go gamer builds to delete their body fast enough because if they press it, chasing headslug is nearly impossible unless its in a big hallway with no objects and doors.
Limiting its uses, forces changeling to absorb someone to regain headslug increases hard of spamming it, which is good.
Its now available to everyone to not make people spend money for one-use ability aswell as making it impossible to not have combat revive ability.
Combat reviving is now less powerful which was requested by Matt balance team member. The only real combat revive now is headslug which is changed above.

## Testing

Compiled, headslugged, absorbed mindless, absorbed target with mind

## Changelog
:cl:
tweak: Headslug is now available to all changelings, but can be done only once per absorption.
tweak: Changelings abilities cannot be used 10 seconds after reviving with stasis.
/:cl:

### CODE REVIEW

**Comment by 1080pCat** [comment]
This feels like a major 'i ded pls nerf' pr.
While i agree that it needs to be nerfed, this doesn't feel like its way too harsh simply because its meta right now, i'd personally rather the headslug lose the ability to go under doors instantly, i believe marm mentioned something about a timer based thing, like contort body has.
And also require a A. living host full stop. or B. require a carbon dead or alive. no monkey gaming.
ALSO ALSO, the headslug shouldn't be able to use the same dead body more than once if the B route is chosen.

**Comment by Danchi299** [comment]
> Making it available to all changelings removes noobtrap with new lings not bying it.

well that's weird reasoning but ok

**Comment by Coolrune206** [comment]
The ability needs a nerf but I don't think this is the way to go about it. What 1080p mentioned about timers would work better, or removing the hide feature altogether (it has speed and better stun/delimbing now, it can lose the ability to hide under shit and slide under doors).

**Comment by matttheficus** [comment]
If Changelings get free headslugs, they need to lose something else - I've been considering making revive put your Changeling Spells on a cooldown (20-30 seconds, perhaps) to stomp out what I dub "combat revives".

If Security downs you, you should be dead barring headslug escape.

**Comment by HMBGERDO** [comment]
Added https://github.com/ParadiseSS13/Paradise/pull/24022 changes to this PR per Matt's permission
Made headslug rechargable after absorbing target with mind
Need to change ability description to fit condition that you need to absorb target with mind

**Comment by Dafrek** [comment]
Headslug in its current state should not become a core part of the changeling kit. Making it free but only 1 use that is replenished by absorbing will probably prevent the worst of it being spammed, but it really isn't a nerf. For people who are robust enough to only need to headslug once in a while its essentially a buff, as you no longer have to spend any points on last resort.

File: code/modules/antagonists/changeling/powers/absorb.dm [hunk 1]
```diff
@@ -86,6 +86,10 @@
 			target_cling.absorbed_dna.len = 1
 			target_cling.absorbed_count = 0
 
+		if(cling.headslugged)
+			cling.headslugged = FALSE
+			to_chat(user, "<span class='boldnotice'>Absorbing [target] regained our ability to headslug.</span>")
```
Comment (lewcc) [inline code/modules/antagonists/changeling/powers/absorb.dm:91]:
```suggestion
			to_chat(user, "<span class='boldnotice'>By absorbing [target], we are once again strong enough to turn into a headslug.</span>")
```

**Review by lewcc — COMMENTED** [review]
I still feel a little iffy about the descriptions explicitly stating headslug (since it looks like some of the helptext shies away from that term) but the code mostly looks good beyond that.

File: code/modules/antagonists/changeling/powers/become_headslug.dm [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
-	power_type = CHANGELING_PURCHASABLE_POWER
-	category = /datum/changeling_power_category/defence
+	power_type = CHANGELING_INNATE_POWER
 
 /datum/action/changeling/headslug/try_to_sting(mob/user, mob/target)
-	if(tgui_alert(user, "Are you sure you wish to do this? This action cannot be undone.", "Sting", list("Yes", "No")) == "No")
+	if(cling.headslugged)
+		to_chat(user, "<span class='notice'>We need to absorb a humanoid to headslug again.</span>")
 		return
+	if(tgui_alert("Are you sure you wish to do this? This action cannot be undone.",,"Yes","No") == "No")
```
Comment (S34NW) [inline code/modules/antagonists/changeling/powers/become_headslug.dm:16]:
```suggestion
	if(tgui_alert("Are you sure you wish to do this? This action cannot be undone.",,"Yes","No") != "Yes")
```

Comment (S34NW) [reply code/modules/antagonists/changeling/powers/become_headslug.dm:16]: LGTM after the above safety

**Review by S34NW — APPROVED** [review]

**Pull request merged**